lib/config/routes.dart

```diff
@@ -46,77 +48,117 @@ class AppRoutes {
-            ),
-            VWidget(
-              path: '/stories/:roomid',
-              widget: const StoryPage(),
-              stackedRoutes: [
-                VWidget(
-                  path: 'share',
-                  widget: const AddStoryPage(),
-                ),
-              ],
-            ),
-            VWidget(
-              path: '/spaces/:roomid',
-              widget: const ChatDetails(),
-              stackedRoutes: _chatDetailsRoutes,
-            ),
-            VWidget(
-              path: ':roomid',
-              widget: const Chat(),
-              stackedRoutes: [
-                VWidget(
-                  path: 'encryption',
-                  widget: const ChatEncryptionSettings(),
-                ),
-                VWidget(
-                  path: 'invite',
-                  widget: const InvitationSelection(),
-                ),
-                VWidget(
-                  path: 'details',
-                  widget: const ChatDetails(),
-                  stackedRoutes: _chatDetailsRoutes,
-                ),
-              ],
-            ),
-            VWidget(
-              path: '/settings',
-              widget: const Settings(),
-              stackedRoutes: _settingsRoutes,
-            ),
-            VWidget(
-              path: '/archive',
-              widget: const Archive(),
-              stackedRoutes: [
-                VWidget(
-                  path: ':roomid',
-                  widget: const Chat(),
-                  buildTransition: _dynamicTransition,
-                ),
-              ],
-            ),
-            VWidgetWithDependency(
-              di: ContactDI(),
-              path: '/newprivatechat',
-              widget: const NewPrivateChat(),
-            ),
-            VWidgetWithDependency(
-              di: ContactDI(),
-              path: '/newgroup',
-              widget: const NewGroup(),
-            ),
-          ],
-        ),
-      ];
+        VRouter(routes: [
+          VNester(
+            key: const ValueKey("rooms"),
+            path: '/rooms',
+            widgetBuilder: (child) => HomeScreen(child: child,),
+            buildTransition: _leftToRightTransition,
+            nestedRoutes: [
+              VWidget(
+                path: null, 
+                widget: const ChatList(),
+                buildTransition: _bottomToTopTransition,
+              ),
+              VWidgetWithDependency(
+                path: '/contactsTab', 
+                widget: const ContactsTab(),
+                di: ContactDI(),
+                buildTransition: _bottomToTopTransition,
+              ),
+              VWidget(
+                path: '/stories', 
+                widget: const Scaffold(),
+                buildTransition: _bottomToTopTransition,
+              ),
+            ],
+          ),
+          VPopHandler(
+            onPop: (vRedirector) async {
+              vRedirector.to('/rooms');
```

Comment:
you mean: whenever you `pop` in the widget inside, chat list will be open again?